js/repeater.js

```diff
@@ -638,15 +638,12 @@
 					bottom: this.$viewport.css('margin-bottom'),
 					top: this.$viewport.css('margin-top')
 				};
-
-				var staticHeightValue = (staticHeight === 'true' || staticHeight === true) ? this.$element.height() : parseInt(staticHeight, 10);
-				var headerHeight = this.$element.find('.repeater-header').outerHeight();
-				var footerHeight = this.$element.find('.repeater-footer').outerHeight();
-				var bottomMargin = (viewportMargins.bottom === 'auto') ? 0 : parseInt(viewportMargins.bottom, 10);
-				var topMargin = (viewportMargins.top === 'auto') ? 0 : parseInt(viewportMargins.top, 10);
-				height = staticHeightValue - headerHeight - footerHeight - bottomMargin - topMargin;
-
-				this.$viewport.outerHeight(height); // but WHY are we setting the outerHeight of the viewport to this???
+				height = ((staticHeight === 'true' || staticHeight === true) ? this.$element.height() : parseInt(staticHeight, 10)) -
+					this.$element.find('.repeater-header').outerHeight() -
+					this.$element.find('.repeater-footer').outerHeight() -
+					((viewportMargins.bottom === 'auto') ? 0 : parseInt(viewportMargins.bottom, 10)) -
+					((viewportMargins.top === 'auto') ? 0 : parseInt(viewportMargins.top, 10));
```

Comment:
This giant expression needs to be broken out into clarifying variables.  The rewrite code that you delete was much easier to read and this needs to be unpacked the same way.